pkg/ingress/parser.go

```diff
@@ -335,18 +343,55 @@ func (c *controller) generateConfig() error {
 
 	td.HTTPService = make([]*haproxy.HTTPService, 0)
 	td.TCPService = make([]*haproxy.TCPService, 0)
-	tp80 := false
 
-	type httpKey struct {
-		Address    string
-		Port       int
+	type hostBinder struct {
+		Address string
+		Port    int
+	}
+	type httpInfo struct {
 		NodePort   int32
 		OffloadSSL bool
+		Hosts      map[string][]*haproxy.HTTPPath
 	}
-	httpServices := make(map[httpKey][]*haproxy.HTTPPath)
+	httpServices := make(map[hostBinder]*httpInfo)
+	tcpServices := make(map[hostBinder]*haproxy.TCPService)
 	for _, rule := range c.Ingress.Spec.Rules {
 		if rule.HTTP != nil {
-			httpPaths := make([]*haproxy.HTTPPath, 0)
+			binder := hostBinder{Address: rule.HTTP.Address}
+			offloadSSL := false
+
+			if _, foundTLS := c.Ingress.FindTLSSecret(rule.Host); foundTLS && !rule.HTTP.NoTLS {
+				offloadSSL = true
+				if port := rule.HTTP.Port.IntValue(); port > 0 {
+					binder.Port = port
+				} else {
+					binder.Port = 443
+				}
+			} else {
+				offloadSSL = false
+				if port := rule.HTTP.Port.IntValue(); port > 0 {
+					binder.Port = port
+				} else {
+					binder.Port = 80
+				}
+			}
+
+			info := &httpInfo{Hosts: make(map[string][]*haproxy.HTTPPath)}
+			if v, ok := httpServices[binder]; ok {
+				info = v
+			} else {
+				httpServices[binder] = info
+			}
+			info.OffloadSSL = offloadSSL
```

Comment:
Can't user define two different rules with same host, address and port but different paths?
In such case what if one rule has specified `NoTLS` but other rule don't?

Reply:
User can define that but it is not supported by HAProxy. The offloadSSL-ing happens for `addr;port` bind line.